### ISSUE TITLE
Overriding Bootstrap/other style library styles

### ISSUE THREAD

**Issue by skitterm** [css-modules/css-modules#128]
I'm planning on using Bootstrap and CSS Modules on a project. One issue I see is how to override a bootstrap style rule with one in CSS Modules. Since CSS modules only provide one class to a style rule, specificity would indicate that I can't override a bootstrap style rule with more than just one class with a CSS Modules one. Most styles could be overridden (since most are just using one class name), but some could not. Example: 

Bootstrap style:
```css
.dropdown-menu > li > a {
  /* some sample style */
  color: blue;
}
```

CSS Modules-outputted style:
```css
/* won't override due to specificity */
.my-app___auto__genereated__unique-classname {
  color: red;
}
```

I realize that overriding by IDs or using <code>!important</code> is a bad idea, but how do you override styles from other libraries that use two classnames, or a classname and a tag selector? I realize I could write some global css to override, but I obviously want to stay away from that (that's why I'm using CSS Modules).

**Comment by MatteoWebDesigner** [comment]
Yes this is a problem even for me.

**Comment by dounan** [comment]
I haven't used the library yet and am not too familiar with it, but came across this other issue that might be helpful: https://github.com/css-modules/css-modules/issues/118

**Comment by SohKai** [comment]
A way to handle this is by duplicating the specificity chain and adding your local class to the last element, something like `:global(.dropdown-menu > li > a).your-class`. I would say this is better than using `!important` and global overrides, but it's still a hack and doesn't work if you'd like to use `compose`. It could be made slightly better by using mixins to provide the specificity chain, but all-in-all not ideal.

The root cause of this is using a library that has higher specificity than you'd like (especially with something like `.dropdown-menu > li > a`, but it's probable we'll still be working with libraries like this for the foreseeable future. However, given the constraints with `compose` only working with a single local class, I'm not sure there's a better solution.

**Comment by geelen** [comment]
Yep, this is a problem. One of the core problems that lead to CSS Modules, actually, which was that specificity escalation is a one-way process, the only winning move is not to play. So CSS Modules lets you use single-purpose, non-specific classes everywhere, but unfortunately that's incompatible with things like `dropdown-menu > li > a`. 

I had an old idea about "safely" importing something like bootstrap, maybe take a look there? https://github.com/css-modules/css-modules/issues/17

Closing this in favour of #65 which I'll close once I've documented the issue/solutions properly.
